test/smoke/test_smoke.py

```diff
@@ -0,0 +1,40 @@
+# Copyright 2022 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Learn more about testing at: https://juju.is/docs/sdk/testing
+
+import logging
+
+from pytest_operator.plugin import OpsTest
+
+logger = logging.getLogger(__name__)
+
+
+CURRENT_CHANNEL = '20.04'  # Focal
+
+
+async def test_smoke(ops_test: OpsTest):
+    # Verify that we can deploy charms from supported series.
+
+    # Build the charm. (We just build it for focal -- it *should* work to deploy it on
+    # older versions of Juju.)
+    charm = await ops_test.build_charm("./test/charms/test_smoke/")
+
+    for series in ['focal', 'bionic', 'xenial']:
```

Comment:
use `pytest.mark.parametrize` to give a better error message (use `ids=['focal', ...]` for bonus points)

Reply:
I started in on this, and it ended up adding a lot of complexity. The build needs to get moved into a separate routine, and then we need a data structure to track our charm artifact. None of that is difficult to write, but it didn't feel worthwhile to do so until this test gets more complex.